aries_cloudagent/vc/data_integrity/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,210 @@
+        assert (
+            options.cryptosuite == "eddsa-jcs-2022"
+        ), "Expected proof.cryptosuite to be `eddsa-jcs-2022`"
+
+        return self._canonicalize(unsecured_document)
+
+    def hashing(self, transformed_document: bytes, canonical_proof_config: bytes):
+        """Hashing algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#hashing-eddsa-jcs-2022.
+        """
+        return (
+            sha256(canonical_proof_config).digest()
+            + sha256(transformed_document).digest()
+        )
+
+    async def proof_serialization(
+        self, hash_data: bytes, options: DataIntegrityProofOptions
+    ):
+        """Proof Serialization Algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#proof-serialization-eddsa-jcs-2022.
+        """
+        # TODO encapsulate in a key manager method
+        if options.verification_method.startswith("did:key:"):
+            multikey = options.verification_method.split("#")[-1]
+            key_info = await self.key_manager.from_multikey(multikey)
+
+        else:
+            key_info = await self.key_manager.from_kid(options.verification_method)
+
+        return await self.wallet.sign_message(
+            message=hash_data,
+            from_verkey=multikey_to_verkey(key_info["multikey"]),
+        )
+
+    def _canonicalize(self, data: dict):
+        """Json canonicalization."""
+        return canonicaljson.encode_canonical_json(data)
+
+    async def verify_proof(self, secured_document: dict):
+        """Verify proof algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#verify-proof-eddsa-jcs-2022.
+        """
+        unsecured_document = secured_document.copy()
+        proof = unsecured_document.pop("proof")
+        proof_options = proof.copy()
+        proof_bytes = multibase.decode(proof_options.pop("proofValue"))
+
+        try:
+            # Currently leaving context processing out of scope,
+            # leaving code commented as it's technically an algorithm step.
+            # Due to the cryptosuite being based on JSON canonicalization,
+            # the integrity of the document is protected without RDF processing.
+
+            # https://www.w3.org/TR/vc-data-integrity/#validating-contexts
+
+            # assert secured_document['@context'] == proof_options['@context']
+            # unsecured_document['@context'] = proof_options['@context']
+
+            proof_options = DataIntegrityProofOptions.deserialize(proof_options)
+            transformed_data = self.transformation(unsecured_document, proof_options)
+            proof_config = self.proof_configuration(proof_options)
+            hash_data = self.hashing(transformed_data, proof_config)
+            verified = await self.proof_verification(
+                hash_data, proof_bytes, proof_options
+            )
+            if not verified:
+                raise CryptosuiteError("Invalid signature.")
+
+        except (AssertionError, CryptosuiteError) as err:
```

Comment:
As I mentioned over Discord the other day, catching an AssertionError is a red flag. Can you tell me about why we expect to catch AssertionErrors here?

See this discussion from ChatGPT:

> Catching an `AssertionError` in Python is generally considered an anti-pattern because assertions are meant to serve as a development-time tool, not as a mechanism for handling run-time errors. Here’s why:
> 
> 1. **Assertions are for debugging, not control flow:**
>    - Assertions are intended to catch bugs during development by verifying that conditions that "should never happen" are true. If an assertion fails, it indicates a flaw in the code that needs to be fixed. Catching the `AssertionError` implies that the code expects such situations to occur, which defeats the purpose of the assertion.
>    
> 2. **Assertions can be disabled:**
>    - Python allows assertions to be disabled in optimized mode (using the `-O` flag when running Python). When this happens, the `assert` statements are not even executed, meaning that any `AssertionError` handling code will be skipped. This can lead to inconsistent behavior when running the code in different environments.
>    
> 3. **Confusing intent:**
>    - Catching `AssertionError` may lead to confusion about the intention of the code. An assertion failure should be a sign that something has gone wrong, and attempting to handle it programmatically can obscure the root cause of the problem.
>    
> 4. **Better alternatives exist:**
>    - If you need to handle predictable or recoverable conditions, it’s better to use explicit checks (e.g., `if` statements) and raise appropriate exceptions, such as `ValueError` or `TypeError`. This approach makes the code more readable and predictable, as it clearly distinguishes between code that handles errors and code that verifies assumptions during development.
> 
> For example, consider this anti-pattern:
> 
> ```python
> try:
>     assert some_condition
> except AssertionError:
>     handle_error()
> ```
> 
> A better approach would be to explicitly check the condition:
> 
> ```python
> if not some_condition:
>     handle_error()
> ```
> 
> This makes it clear that the condition is part of the program's expected flow, rather than an assertion that something has gone fundamentally wrong.
> 
> In summary, assertions are designed to catch programmer errors during development and testing, not for error handling in production code. Catching `AssertionError` is therefore discouraged because it misuses assertions as part of control flow.

Reply:
removed the assertion catch